test/Interop/Cxx/class/destructor-correct-abi-irgen.swift

```diff
@@ -0,0 +1,28 @@
+// RUN: %swift -module-name Swift -target x86_64-apple-macosx10.9 -I %S/Inputs -enable-cxx-interop -emit-ir %s -parse-stdlib -parse-as-library | %FileCheck %s -check-prefix=ITANIUM_X64
+// RUN: %swift -module-name Swift -target armv7-none-linux-androideabi -dump-clang-diagnostics -I %S/Inputs -enable-cxx-interop -emit-ir %s -parse-stdlib -parse-as-library -disable-legacy-type-info | %FileCheck %s -check-prefix=ITANIUM_ARM
+// RUN: %swift -module-name Swift -target x86_64-unknown-windows-msvc -dump-clang-diagnostics -I %S/Inputs -enable-cxx-interop -emit-ir %s -parse-stdlib -parse-as-library -disable-legacy-type-info | %FileCheck %s -check-prefix=MICROSOFT_X64
```

Comment:
Using the `parse-stdlib` mode is can interfere with how the importer works, so I'm reluctant to use it in these tests.

Reply:
Without `parse-stdlib` we can't cross-compile to other targets. Because this is testing the ABI it needs to compile with multiple targets. Maybe there's another flag we can use, though?

Reply:
Does it make sense to rely on the various CI jobs to have this covered, and either add regexes or copy this test for each relevant platform? This way we would avoid the `parse-stdlib` mode. It's also quite common to do it this way in other places.

Reply:
Yeah, I'll turn this into a regex. It would be really great if we could just cross-compile the standard library, though. Maybe one day.